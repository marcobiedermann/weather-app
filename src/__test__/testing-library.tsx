import { RenderOptions, render } from '@testing-library/react';
import { ReactElement, ReactNode } from 'react';
import { I18nextProvider } from 'react-i18next';
import { MemoryRouter } from 'react-router-dom';
import i18n from './i18n';

interface ProvidersProps {
  children: ReactNode;
}

function Providers(props: ProvidersProps) {
  const { children } = props;

  return (
    <I18nextProvider i18n={i18n}>
      <MemoryRouter>{children}</MemoryRouter>
    </I18nextProvider>
  );
}

function customRender(ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>) {
  return render(ui, {
    wrapper: Providers,
    ...options,
  });
}

export { customRender as render };
