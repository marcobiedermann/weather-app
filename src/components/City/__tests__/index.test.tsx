import { composeStory } from '@storybook/react';
import { describe, expect, it } from 'vitest';
import { render } from '../../../test/testing-library';
import Meta, { Default } from '../City.stories';

describe('components/City', () => {
  it('renders correctly', () => {
    const ComposedDefault = composeStory(Default, Meta);
    const { asFragment } = render(<ComposedDefault />);

    expect(asFragment()).toMatchSnapshot();
  });
});
