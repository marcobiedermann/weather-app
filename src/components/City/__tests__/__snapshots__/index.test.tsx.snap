// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`components/City > renders correctly 1`] = `
<DocumentFragment>
  <div
    clouds="[object Object]"
    coord="[object Object]"
    id="2950159"
    sys="[object Object]"
    visibility="10000"
    wind="[object Object]"
  >
    <h1>
      Berlin
    </h1>
    2020-11-15 10:09
    <div>
      Temp: 12.61
    </div>
    <ul>
      <li>
        clear sky
        <img
          alt="clear sky"
          src="https://openweathermap.org/img/w/01d.png"
        />
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
